scrapy_poet/injection.py

```diff
@@ -303,7 +304,14 @@ def get_callback(request, spider):
     return request.callback
 
 
-def is_callback_requiring_scrapy_response(callback: Callable):
+def get_response_annotation(callback: Callable):
```

Comment:
I think `get_response_signature_parameter` is more apt since it's returning `inspect.Parameter`.

We can keep this public, as it's quite handy for **scrapy-poet** users to also have out-of-the-box response introspection if they need it. I also don't see any reason to make it private as well.